<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.22.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Microbial_pipeline.r - DORCIERR</title>
<meta name="description" content="``` Script Written by Zach Quinlan 06/19/19 Re-organization of DORCIERR_FCM_fDOM.R because it needs to be cleaner 07/15/2019 Only working on daytime exudation and remineralization Rewritten with changes to RR3 starting pipeline 10/11/2019 16s rRNA amplicon seque3nce data added in a upaded 10/18/2019 Added in ClassyFire annotations from Inchi_keys 10/07/2019 This has been rewritten for the new pipeline 12/18/2019 This was split to just look at metabolite trends on 05/12/2020">


  <meta name="author" content="Zach Quinlan">
  
  <meta property="article:author" content="Zach Quinlan">
  


<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="DORCIERR">
<meta property="og:title" content="Microbial_pipeline.r">
<meta property="og:url" content="/%7B%7B%20site.baseurl%20%7D%7D/contents/Microbial_pipeline.R/">


  <meta property="og:description" content="``` Script Written by Zach Quinlan 06/19/19 Re-organization of DORCIERR_FCM_fDOM.R because it needs to be cleaner 07/15/2019 Only working on daytime exudation and remineralization Rewritten with changes to RR3 starting pipeline 10/11/2019 16s rRNA amplicon seque3nce data added in a upaded 10/18/2019 Added in ClassyFire annotations from Inchi_keys 10/07/2019 This has been rewritten for the new pipeline 12/18/2019 This was split to just look at metabolite trends on 05/12/2020">







  <meta property="article:published_time" content="2021-05-07T18:04:28-07:00">





  

  


<link rel="canonical" href="/%7B%7B%20site.baseurl%20%7D%7D/contents/Microbial_pipeline.R/">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": "Zach Quinlan",
      "url": "/%7B%7B%20site.baseurl%20%7D%7D/"
    
  }
</script>







<!-- end _includes/seo.html -->



  <link href="/%7B%7B%20site.baseurl%20%7D%7D/feed.xml" type="application/atom+xml" rel="alternate" title="DORCIERR Feed">


<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/%7B%7B%20site.baseurl%20%7D%7D/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">

<!--[if IE]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->


    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--posts wide">
    <nav class="skip-links">
  <h2 class="screen-reader-text">Skip links</h2>
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    

<div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        
          <a class="site-logo" href="/%7B%7B%20site.baseurl%20%7D%7D/"><img src="/%7B%7B%20site.baseurl%20%7D%7D/assets/images/logo.png" alt="DORCIERR"></a>
        
        <a class="site-title" href="/%7B%7B%20site.baseurl%20%7D%7D/">
          DORCIERR
          
        </a>
        <ul class="visible-links"><li class="masthead__menu-item">
              <a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/">Repository Contents</a>
            </li><li class="masthead__menu-item">
              <a href="/%7B%7B%20site.baseurl%20%7D%7D/assets/images/album">Photos</a>
            </li><li class="masthead__menu-item">
              <a href="https://www.doi.org/">Paper</a>
            </li></ul>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Repository Contents</span>
        

        
        <ul>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/canopus_model.R/">canopus_model.R</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/bicluster.py/">bicluster.py</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/Microbial_pipeline.R/">Microbial_pipeline.R</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/metabolite_modeled_lability.R/">metabolite_modeled_lability.R</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/12182019DORCIERR_working_pipeline.R/">12182019DORCIERR_working_pipeline.R</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/metabolite_trends.R/">metabolite_trends.R</a></li>
          
            <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/_pages/contents/current_dorcierr_working_pipeline.R/">current_dorcierr_working_pipeline.R</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <a href="/%7B%7B%20site.baseurl%20%7D%7D/assets/images/album"><span class="nav__sub-title">Photos</span></a>
        

        
      </li>
    
      <li>
        
          <a href="https://www.doi.org/"><span class="nav__sub-title">Paper</span></a>
        

        
      </li>
    
  </ul>
</nav>

    
  
  </div>



  <div class="archive">
    
      <h1 id="page-title" class="page__title">Microbial_pipeline.r</h1>
    
    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Script Written by Zach Quinlan 06/19/19
# Re-organization of DORCIERR_FCM_fDOM.R because it needs to be cleaner 07/15/2019
# Only working on daytime exudation and remineralization
# Rewritten with changes to RR3 starting pipeline 10/11/2019
# 16s rRNA amplicon seque3nce data added in a upaded 10/18/2019
# Added in ClassyFire annotations from Inchi_keys 10/07/2019
# This has been rewritten for the new pipeline 12/18/2019
# This was split to just look at metabolite trends on 05/12/2020


# LOADING -- packages -------------------------------------------------------
#Data mungering
library(tidyverse)
library(data.table)
library(DescTools)
library(broom)
library(readxl)
library(multcomp)
library(CHNOSZ)
library(furrr)
library(future)
library(biclustermd)
library(webchem)
library(classyfireR)
library(randomForest)
library(ggpubr)
library(rsq)

#PCoA, PERMANOVA
library(vegan)
library(ape)

#Visualizations
library(wesanderson)
library(RColorBrewer)
library(gplots)
library(gtable)
library(ggnewscale)

#Defining functions and removing issues with overlapping function calls
map &lt;- purrr::map
select &lt;- dplyr::select
tidy &lt;- broom::tidy
rename &lt;- dplyr::rename
mutate &lt;- dplyr::mutate

zscore &lt;- function(x) {
  (x-mean(x, na.rm = TRUE))/sd(x, na.rm = TRUE)
}


# LOADING -- Dataframes ---------------------------------------------------
## FCM
dorc_fcm_fdom &lt;- read_xlsx("~/Documents/GitHub/DORCIERR/data/raw/DOC_fDOM_FCM/DORCIERR_fDOM_FCM.xlsx")%&gt;%
  rename(sample_name =`Sample Name of DORCIERR_FCM_Final`)%&gt;%
  rename('DayNight' = 'Experiment')%&gt;%
  rename(Organism = 'Organsim')%&gt;%
  mutate(Organism = case_when(Organism == "Water" ~ "Water control",
                              TRUE ~ as.character(Organism)))

# 16s rRNA sequences
microbe_abundance_raw &lt;- read_tsv("~/Documents/GitHub/DORCIERR/data/raw/microbes/MCR2017.16S.Nelson.Pipeline.October2019/abundance_table_100.shared.tsv")
microbe_taxonomy &lt;- read_tsv("~/Documents/GitHub/DORCIERR/data/raw/microbes/MCR2017.16S.Nelson.Pipeline.October2019/annotations_100.taxonomy.tsv")

# PRE-STATS CLEANING -- Microbes and pre-filtering OTUs for abundance-----------------------------------------------
microbe_combined &lt;- microbe_abundance_raw%&gt;%
  dplyr::select(-1)%&gt;%
  mutate(Group = case_when(Group == "Dorcierr_D_DT_1_TFD" ~ "D_DT_1_TFD",
                           Group == "DORCIERR_D_WA_2_TFN" ~ "D_WA_2_TFN",
                           Group == "D_PV_2_TFN_SA504_SC704" ~ "D_PV_2_TFN",
                           Group == "D_PV_2_TFN_SA503_SC704" ~ "D_PV_3_TFN",
                           Group == "D_WA_4_TFN_SA503_SC703" ~ "D_PL_3_TFN",
                           Group == "D_WA_4_TFN_SA504_SC703" ~ "D_WA_4_TFN",
                           TRUE ~ as.character(Group)))%&gt;%
  filter(Group %like% "%D_%")%&gt;%
  rename(sample_code = Group)%&gt;%
  gather(OTU, reads, 3:ncol(.))%&gt;%
  # left_join(., microbe_taxonomy, by = "OTU")%&gt;%
  # dplyr::select(-Size)%&gt;%
  # separate(Taxonomy, c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "OTU_id"), sep = ";", remove = FALSE)%&gt;%
  separate(sample_code, c("Experiment", "Organism", "Replicate", "Timepoint"), sep = "_", remove = FALSE)%&gt;%
  dplyr::mutate(Experiment = case_when(Experiment == "D" ~ "dorcierr",
                                       Experiment == "M" ~ "mordor",
                                       Experiment == "R" ~ "RR3",
                                       TRUE ~ as.character(Experiment)))%&gt;%
  dplyr::mutate(Organism = case_when(Organism == "CC" ~ "CCA",
                                     Organism == "DT" ~ "Dictyota",
                                     Organism == "PL" ~ "Porites lobata",
                                     Organism == "PV" ~ "Pocillopora verrucosa",
                                     Organism == "TR" ~ "Turf",
                                     Organism == "WA" ~ "Water control",
                                     Organism == "IN" ~ "Influent",
                                     Organism == "OF" ~ "Offshore",
                                     TRUE ~ as.character(Organism)))%&gt;%
  separate(Timepoint, c("Timepoint", "DayNight"), sep = 2)%&gt;%
  mutate(DayNight = case_when(DayNight == "D" ~ "Day",
                              TRUE ~ "Night"))%&gt;%
  filter(Organism != "Offshore",
         Organism != "Influent")%&gt;%
  mutate(reads = case_when(reads == 0 ~ 1/numOtus,
                           TRUE ~ as.numeric(reads)))%&gt;%
  # left_join(fcm_wdf%&gt;%
  #             rename(sample_code = sample_name)%&gt;%
  #             select(c(`Cells µL-1`, sample_code)), by = "sample_code")%&gt;%
  group_by(sample_code)%&gt;%
  mutate(ra = reads/sum(reads))%&gt;%
  # cell_abun = ra*`Cells µL-1`,
  # log10 = log10(cell_abun + 0.001))%&gt;%
  ungroup()%&gt;%
  group_by(OTU)%&gt;%
  mutate(abundant = case_when(max(ra) &gt; 0.01 | sum(ra &gt; 0.001) &gt;= 3 ~ "abundant",
                              TRUE ~ "rare"))

microbe_no_rare &lt;- microbe_combined%&gt;%
  select(-c(sample_code, reads, numOtus))%&gt;%
  ungroup()%&gt;%
  filter(Timepoint == "TF")%&gt;%
  group_by(OTU, DayNight, Organism)%&gt;%
  filter(abundant == "abundant")%&gt;%
  mutate(asin = asin(sqrt(ra)))%&gt;%
  select(-c(ra, abundant))


# PRE-STATS CLEANING -- microbe RA data  --------------------------------------------
ra_bigger_TF &lt;- microbe_combined%&gt;%
  filter(abundant == "abundant",
         Timepoint == 'TF')%&gt;%
  select(-c(abundant, reads, numOtus, sample_code, Replicate))%&gt;%
  group_by(Organism, DayNight, OTU)%&gt;%
  summarize_if(is.numeric, mean)%&gt;%
  spread(Organism, ra)%&gt;%
  gather(Organism, ra, 3:7)%&gt;%
  mutate(difference = ra - `Water control`)%&gt;%
  filter(difference &gt; 0)%&gt;%
  select(DayNight, OTU, Organism)

average_ra &lt;- microbe_no_rare%&gt;%
  select(-c(Experiment, Replicate))%&gt;%
  group_by(OTU, Organism, DayNight, Timepoint)%&gt;%
  summarize_if(is.numeric, mean, na.rm = TRUE)%&gt;%
  ungroup()

# PRE-STATS CLEAINING -- FCM Stats prep---------------------------------------------------------------------
## This should calculate mean cells per hour per µL for each organism
## Just looks at log10(TF) - mean(log10(T0))
## Will result in log10(cells*µL-1)*hr^-1
fcm_wdf &lt;- dorc_fcm_fdom%&gt;%
  dplyr::select(c(1:8,34:ncol(.)))

fcm_th_t0_prep &lt;-fcm_wdf%&gt;%
  filter(Timepoint %like any% c('T0', 'T4'),
         !Organism == 'Influent',
         !Organism == 'Offshore')%&gt;%
  mutate(log_10_cellsµL = log10(`Cells µL-1`))%&gt;%
  dplyr::select(c(5:8, 15))%&gt;%
  spread(Timepoint, log_10_cellsµL)

fcm_t0 &lt;- fcm_th_t0_prep%&gt;%
  dplyr::select(-c(T4, Replicate))%&gt;%
  group_by(Organism, DayNight)%&gt;%
  summarize_if(is.numeric, mean, na.rm = TRUE)

fcm_rate_th_t0 &lt;- fcm_th_t0_prep%&gt;%
  dplyr::select(-T0)%&gt;%
  left_join(., fcm_t0, by = c("Organism", "DayNight"))%&gt;%
  add_column(change_per_hour = (.$T4 - .$T0)/24)

fcm_t7 &lt;- fcm_wdf%&gt;%
  mutate(log_10_cellsµL = log10(`Cells µL-1`))%&gt;%
  dplyr::select(c(5:8, 15))%&gt;%
  filter(Timepoint == c('TF'),
         !Organism == 'Influent')%&gt;%
  spread(Timepoint, log_10_cellsµL)

## This is the actual dataframe to use for running FCM stats
fcm_stats_df &lt;- left_join(fcm_t7, fcm_rate_th_t0, by = c("Organism", "Replicate", "DayNight"))%&gt;%
  dplyr::select(-c(5,6))%&gt;%
  gather(test, log_value, 4:5)


# SET SEED ----------------------------------------------------------------
set.seed(2005)

# STATS ANOVA -- Microbe TWO-Way ------------------------------------------
aov_microbe &lt;- microbe_no_rare%&gt;%
  group_by(OTU)%&gt;%
  nest()%&gt;%
  mutate(anova = map(data, ~ aov(asin ~ Organism*DayNight, .x)%&gt;%
                       tidy()%&gt;%
                       filter(!term == "Residuals")%&gt;%
                       dplyr::select(term, p.value)))%&gt;%
  dplyr::select(-data)%&gt;%
  unnest(anova)

anova_microbe_pvalues &lt;- aov_microbe%&gt;%
  ungroup()%&gt;%  
  filter(p.value &lt; 0.05)%&gt;%
  add_column(FDR = p.adjust(.$p.value, method = "BH"))%&gt;%
  filter(FDR &lt; 0.05)

organism_significant_microbes &lt;- as.vector(anova_microbe_pvalues%&gt;%
                                             filter(term == "Organism"))$OTU

DayNight_significant_microbes &lt;- as.vector(anova_microbe_pvalues%&gt;%
                                             filter(term != "Organism"))$OTU
# PRE-POST-HOC CLEANING -- Microbe Dunnetts and DayNight anova -------------------------------
mic_organism_post_hoc &lt;- microbe_no_rare%&gt;%
  filter(OTU %in% organism_significant_microbes)

daynight_microbe_post_hoc &lt;- microbe_no_rare%&gt;%
  filter(OTU %in% DayNight_significant_microbes)

# STATS POST-HOC -- FCM Tukeys ----------------------------------------------------------
# Tukey growth rates for the first half of the expierment
tukey_model_fcm &lt;-fcm_stats_df%&gt;%
  group_by(test, DayNight)%&gt;%
  nest()%&gt;%
  mutate(data = map(data, ~aov(log_value ~ Organism, .x)%&gt;%
                      TukeyHSD(p.adjust.methods = "BH")%&gt;%
                      tidy()))%&gt;%
  unnest(data)%&gt;%
  filter(adj.p.value &lt; 0.05)

# STATS POST-HOC -- MICROBES Dunnetts -----------------------------
organism_order_micro &lt;- as.factor(mic_organism_post_hoc$Organism)%&gt;%
  relevel("Water control")%&gt;%
  levels()%&gt;%
  as.vector()

dunnett_microbe_pvals &lt;- mic_organism_post_hoc%&gt;%
  group_by(DayNight, OTU)%&gt;%
  mutate(sum = sum(asin))%&gt;%
  filter(sum != 0)%&gt;%
  dplyr::select(-sum)%&gt;%
  mutate(Organism = factor(Organism))%&gt;%
  mutate(Organism = fct_relevel(Organism, organism_order_micro))%&gt;%
  nest()%&gt;%
  mutate(dunnett = map(data, ~ aov(asin ~ Organism, .x)%&gt;%
                         glht(linfct = mcp(Organism = "Dunnett"))),
         dunnett_summary = map(dunnett, ~summary(.x)%&gt;%
                                 tidy()))%&gt;%
  dplyr::select(-c(data,dunnett))%&gt;%
  unnest(dunnett_summary)%&gt;%
  dplyr::select(-c(4:7))%&gt;%
  mutate(lhs = gsub(" - Water control", "", lhs))%&gt;%
  rename("Organism" = "lhs")%&gt;%
  ungroup()%&gt;%   
  add_column(FDR = p.adjust(.$p.value, method = "BH"))%&gt;%
  filter(FDR &lt; 0.05)

# STATS POST-HOC -- DayNight MICROBES t-test organism --------------------
daynight_microbe_pvals &lt;- mic_organism_post_hoc%&gt;%
  group_by(Organism, OTU)%&gt;%
  mutate(sum = sum(asin))%&gt;%
  filter(sum != 0)%&gt;%
  dplyr::select(-sum)%&gt;%
  nest()%&gt;%
  mutate(data = map(data, ~ aov(asin ~ DayNight, .x)%&gt;%
                      tidy()))%&gt;%
  unnest(data)%&gt;%
  dplyr::select(-c(4:7))%&gt;%
  filter(term != "Residuals")%&gt;%
  ungroup()%&gt;%   
  add_column(FDR = p.adjust(.$p.value, method = "BH"))%&gt;%
  filter(FDR &lt; 0.05)


# META-STATS -- microbes --------------------------------------------------
dunnett_micro_analysis &lt;- dunnett_microbe_pvals%&gt;%
  dplyr::select(-p.value)%&gt;%
  spread(Organism, FDR)%&gt;%
  add_column(number_exudate_organisms = rowSums(.[3:ncol(.)] &gt;= 0, na.rm = TRUE))%&gt;%
  mutate(microbe_organism = case_when(is.na(CCA) == FALSE &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) &amp;
                                        is.na(`Pocillopora verrucosa`)  &amp;
                                        is.na(`Porites lobata`) ~ "CCA",
                                      is.na(CCA)  &amp; 
                                        is.na(Dictyota) == FALSE &amp;
                                        is.na(Turf) &amp;
                                        is.na(`Pocillopora verrucosa`)  &amp;
                                        is.na(`Porites lobata`) ~ "Dictyota",
                                      is.na(CCA)  &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) == FALSE &amp;
                                        is.na(`Pocillopora verrucosa`)  &amp;
                                        is.na(`Porites lobata`) ~ "Turf",
                                      is.na(CCA) &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) &amp;
                                        is.na(`Pocillopora verrucosa`) == FALSE &amp;
                                        is.na(`Porites lobata`) ~ "Pocillopora verrucosa",
                                      is.na(CCA) &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) &amp;
                                        is.na(`Pocillopora verrucosa`)  &amp;
                                        is.na(`Porites lobata`) == FALSE ~ "Porites lobata",
                                      is.na(`Pocillopora verrucosa`) == FALSE &amp;
                                        is.na(CCA) == FALSE &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf)  |
                                        is.na(`Porites lobata`) == FALSE &amp;
                                        is.na(CCA) == FALSE &amp;
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) ~ "Corraline",
                                      is.na(`Pocillopora verrucosa`) &amp;
                                        is.na(CCA) == FALSE &amp; 
                                        is.na(Dictyota) == FALSE &amp;
                                        is.na(`Porites lobata`) |
                                        is.na(CCA) == FALSE &amp;
                                        is.na(Turf) == FALSE &amp;
                                        is.na(`Porites lobata`) &amp;
                                        is.na(`Pocillopora verrucosa`) ~ "Algae",
                                      is.na(`Dictyota`) &amp;
                                        is.na(CCA)  &amp;
                                        is.na(Turf) &amp;
                                        is.na(`Porites lobata`) == FALSE &amp;
                                        is.na(`Pocillopora verrucosa`) == FALSE ~ "Coral",
                                      is.na(`Dictyota`) == FALSE &amp;
                                        is.na(CCA)  &amp;
                                        is.na(Turf) == FALSE &amp;
                                        is.na(`Porites lobata`) &amp;
                                        is.na(`Pocillopora verrucosa`) ~ "Fleshy Algae",
                                      is.na(CCA)  &amp; 
                                        is.na(Dictyota) &amp;
                                        is.na(Turf) == FALSE &amp;
                                        is.na(`Pocillopora verrucosa`)  &amp;
                                        is.na(`Porites lobata`) ~ "Turf",
                                      number_exudate_organisms &gt; 3 ~ "Primary Producers",
                                      TRUE ~ "Cosmo"))%&gt;%
  left_join(microbe_taxonomy, by = "OTU")

micro_sigs_vector &lt;- as.vector(dunnett_micro_analysis$OTU)


# META-STATS -- Hierarchical cluster matrix----------------------------------------
hc_microbe &lt;- mic_organism_post_hoc%&gt;%
  # filter(OTU %in% rf_microbe_sigs)%&gt;%
  ungroup()%&gt;%
  group_by(OTU)%&gt;%
  mutate(zscore = zscore(log10))%&gt;%
  ungroup()%&gt;%
  dplyr::select(c(Organism, DayNight, Replicate, OTU, zscore))%&gt;%
  unite(sample, c("Organism", "DayNight", "Replicate"), sep = "_")%&gt;%
  left_join(microbe_taxonomy, by = "OTU")%&gt;%
  separate(Taxonomy, c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "OTU_id"), sep = ";")%&gt;%
  unite(OFGO, c("Order", "Family", "Genus", "OTU"), sep = ";")%&gt;%
  select(-c(Size, OTU_id, Kingdom, Phylum, Class))%&gt;%
  spread(OFGO, zscore)

write_csv(hc_microbe%&gt;%
            select(everything(), sample), "~/Documents/GitHub/DORCIERR/data/plots/microbe_hc_df.csv")

# META-STATS -- RA significant dunnetts -----------------------------------
ra_dunnetts &lt;- dunnett_microbe_pvals%&gt;%
  left_join(average_ra, by = c('OTU', 'Organism', "DayNight"))%&gt;%
  left_join(microbe_taxonomy, by = 'OTU')%&gt;%
  separate(Taxonomy, c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "sp"), sep = ";")%&gt;%
  unite(Tax_plot, c("Order", "Family", "Genus", "OTU"), sep = " ", remove = FALSE)

top5_microbes &lt;- ra_dunnetts%&gt;%
  group_by(DayNight, Organism)%&gt;%
  nest()%&gt;%
  mutate(data = map(data, ~ top_n(.x, 5, ra)))%&gt;%
  filter(DayNight == 'Day')%&gt;%
  unnest(data)

top5_microbes%&gt;%
  ggplot(aes(Organism, ra, fill = Tax_plot)) +
  geom_bar(stat = 'identity')

pdf("./plots/org_microbe_ra.pdf", width = 15, height = 10)
average_ra%&gt;%
  left_join(microbe_taxonomy, by = 'OTU')%&gt;%
  separate(Taxonomy, c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "sp"), sep = ";")%&gt;%
  unite(Tax_plot, c("Order", "Family", "Genus", "OTU"), sep = " ", remove = FALSE)%&gt;%
  filter(DayNight == 'Day')%&gt;%
  ggplot(aes(x="", y=ra, fill = Class)) +
  geom_bar(width = 1, size = 0.5, color = "white", stat = "identity") +
  labs(x =NULL, y = NULL, title = "") +
  theme_classic() +
  facet_wrap(~Organism) +
  # scale_fill_manual(values = c('#78B7C5', '#EBCC2A', "#00A08A")) +
  theme(axis.line = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank()) +
  coord_polar("y", start = 0)
dev.off()

</code></pre></div></div>


<ul class="taxonomy__index">
  
  
</ul>





  </div>
</div>
    </div>

    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    

    <ul class="author__urls social-icons">
      
        
          
            <li><a href="" rel="nofollow noopener noreferrer"><i class="fas fa-fw fa-link" aria-hidden="true"></i><span class="label">Website</span></a></li>
          
        
          
            <li><a href="https://www.twitter.com/zquinlan" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-twitter-square" aria-hidden="true"></i><span class="label">Twitter</span></a></li>
          
        
          
            <li><a href="https://www.researchgate.net/profile/zachary-quinlan" rel="nofollow noopener noreferrer"><i class="fab fa-researchgate" aria-hidden="true"></i><span class="label">ResearchGate</span></a></li>
          
        
          
            <li><a href="https://github.com/zquinlan" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-github" aria-hidden="true"></i><span class="label">Github</span></a></li>
          
        
          
            <li><a href="https://orcid.org/" rel="nofollow noopener noreferrer"><i class="fab fa-orcid" aria-hidden="true"></i><span class="label">OrcID</span></a></li>
          
        
      

    
      <li><a href="/%7B%7B%20site.baseurl%20%7D%7D/feed.xml"><i class="fas fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
    
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2021 Zach Quinlan. Powered by <a href = "https://zquinlan.github.io/acio">AcIo</a> &amp; <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/%7B%7B%20site.baseurl%20%7D%7D/assets/js/main.min.js"></script>










  </body>
</html>
